app/controllers/schedule_controller.rb

```diff
@@ -2,33 +2,38 @@
 
 class ScheduleController < ApplicationController
   def save_scheduled_publishing_datetime
-    Edition.find_and_lock_current(document: params[:document]) do |edition|
-      checker = Requirements::ScheduledDatetimeChecker.new(permitted_params)
-      issues = checker.pre_submit_issues
-
-      if issues.any?
-        flash["alert_with_items"] = {
-          title: I18n.t!("requirements.scheduled_datetime.title"),
-          items: issues.items(
-            link_options: {
-              scheduled_datetime: { href: "#scheduled_publishing_datetime" },
-            },
-          ),
-        }
-        flash[:scheduled_publishing_params] = permitted_params
-        redirect_to document_path(edition.document)
-        next
-      end
-
-      set_scheduled_publishing_datetime(edition, checker.parsed_datetime)
-
+    result = Schedule::SaveDatetimeInteractor.call(params: params, user: current_user)
+    edition, issues = result.to_h.values_at(:edition, :issues)
+
+    if issues
+      flash.now["alert_with_items"] = {
+        "title" => I18n.t!("requirements.scheduled_datetime.title"),
+        "items" => issues.items(
+          link_options: {
+            scheduled_datetime: { href: "#scheduled_publishing_datetime" },
```

Comment:
I think this is because we check the date and time combination rather than the separate date and time inputs, so we don't have checks that produce issues for the date and time separately.  This will probably make it difficult to highlight the fields that need correcting.  Would be good to have a chat about how to tackle this.

Reply:
ok cool - just ping me when you get a chance. I'll need some more context.